### ISSUE TITLE
add test to determine if WoERatioCategoricalEncoder returns an error when the probability in the denominator is 0

### ISSUE THREAD

**Issue by solegalli** [feature-engine/feature_engine#56]
At the moment the test is commented out:
https://github.com/solegalli/feature_engine/blob/master/tests/test_encoding/test_woe_encoder.py#L112

the aim is to test this bit of code [here](https://github.com/solegalli/feature_engine/blob/master/feature_engine/encoding/woe.py#L205)

I think the test should work, and I commented out because I changed the error for a warning. But now, we decided to go back to returning an error.

So in short, the aim is to corroborate that the commented out test checks the intended bit of code and if yes, uncomment it, otherwise, replace it by the suitable test.

**Comment by solegalli** [comment]
test cases 11 and 12. I think there is something wrong with how I handle the  values == 0 in the class

**Comment by solegalli** [comment]
@NicoGalli Maybe you can have a look at this one when you separate the WoE encoder from the RatioEncoder?

**Comment by noahjgreen295** [comment]
Sorry I missed the merge on the last PR - was due to work situation. Long story short re that, I have a lot more free time on my hands - at least for now :-) 
Would it be helpful for me to take on some of the backlog priority issues, including this one?

**Comment by solegalli** [comment]
Hi @noahjgreen295 

Nice to hear back from you. No worries at all... I really appreciate your help, so please do keep it coming ;)
